test-data/unit/check-based-nested-typevar.test

```diff
@@ -0,0 +1,11 @@
+[case testNestedTypeVarBound]
+from typing import TypeVar, List
+
+T = TypeVar("T", bound=int)
+L = TypeVar("L", bound=List[T])
+
+def foo(l: L) -> T:
+    return l[0]
+
+reveal_type(foo([True]))  # N: Revealed type is "bool"
+foo([""])  # E: Value of type variable "T" of "foo" cannot be "str"  [type-var]
```

Comment:
does it work on constraints and paramspecs

Reply:
Haven't tried

Reply:
no